tests/jobs/test_scheduler_job.py

```diff
@@ -3793,6 +3793,10 @@ def test_create_dag_runs_datasets(self, session, dag_maker):
         dataset1 = Dataset(uri="ds1")
         dataset2 = Dataset(uri="ds2")
 
+        with dag_maker(dag_id="datasets-consumer-single", schedule=[dataset1]):
```

Comment:
Why is this block being moved?

Reply:
@RNHTTR The DAG needs to be created first before the event can arrive for the DAG to see all the events.